app/controllers/representatives_controller.rb

```diff
@@ -0,0 +1,48 @@
+class RepresentativesController < ApplicationController
+
+  before_action :set_representative, only: [:show, :edit, :update, :destroy]
+
+  before_action :set_type
+
+  def index
+   if params[:search]
+      @representatives = type_class.search(params[:search]).page(params[:page]).per(10)
+    else
+      @representatives = type_class.all.with_gender("FEMININO").page(params[:page]).per(10)
+    end
+
+  end
+
+  def show
+    @representative = type_class.find(params[:id]).decorate
+    if user_signed_in?
+      render 'profile' if current_user.id == @representative.user_id
+    else
+      render 'show'
+    end
+    @proposals = @representative.proposals
+  end
+
+  def representative_params
+    params.require(:representative).permit(:name, :occupation, :gender, :voter_registration)
+  end
+
+  private
+
+    def set_type
+       @type = type
```

Comment:
Se `set_type` for invocado múltiplas vezes durante a resposta a uma solicitação,
```ruby
@type ||= type
```
fará com que `type` só seja invocado se a variável `@type` for `nil` -- o que, nesse contexto, seria muito vantajoso, já que o método `type` faz consulta no banco de dados.

http://stackoverflow.com/questions/995593/what-does-or-equals-mean-in-ruby